noxfile.py

```diff
@@ -0,0 +1,100 @@
+# Copyright 2019 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import nox
+
+
+@nox.session(python="2.7")
+def cover(session):
+    session.install("pytest", "mock", "coverage", "pytest-cov")
+    session.install(".")
+    session.run(
+        "py.test",
+        "--quiet",
+        "--cov=google.cloud.happybase",
+        "--cov=unit_tests",
+        "--cov-config",
+        ".coveragerc",
+        "unit_tests",
+    )
+
+
+@nox.session(python="2.7")
+def docs(session):
+    session.install("pytest", "mock", "Sphinx", "sphinx_rtd_theme")
+    session.install(".")
+    session.run(
+        "python",
+        "-c",
+        "import shutil; shutil.rmtree('docs/_build', ignore_errors=True)",
+    )
+    session.run(
+        "sphinx-build",
+        "-W",
+        "-b",
+        "html",
+        "-d",
+        "docs/_build/doctrees",
+        "docs",
+        "docs/_build/html",
+    )
+    session.run(
+        "python",
+        "scripts/verify_included_modules.py",
+        "--build-root",
+        "_build"
+    )
+
+
+@nox.session(python="3.7")
+def lint(session):
+    """Run linters.
+    Returns a failure if the linters find linting errors or sufficiently
+    serious code quality issues.
+    """
+    session.install("flake8", "black")
+    session.run(
+        "black",
+        "--check",
+        "src",
+        "docs",
+        "unit_tests",
+        "system_tests")
+    session.run("flake8", "google", "tests")
+
+
+@nox.session(python="3.6")
+def blacken(session):
+    session.install("black")
+    session.run(
+        "black",
+        "noxfile.py",
+        "src",
+        "docs",
+        "unit_tests",
+        "system_tests"
+    )
+
+
+@nox.session(python=["2.7", "3.4", "3.5", "3.6", "3.7"])
+def unit(session):
+    session.install("pytest", "mock")
+    session.install(".")
+    session.run("py.test", "--quiet", "unit_tests")
+
+
+@nox.session(python=["2.7", "3.7"])
+def system(session):
+    session.install("pytest", "mock")
+    session.install(".")
+    session.run("py.test", "--quiet", "system_tests/happybase.py")
```

Comment:
Can we move adding the noxfile w/ blacken to the PR which actually applies blackening?

Reply:
sure. I can strip this out of the PR. Though I need it to run the system test. Our existing runner is causing oddness that makes tests that should fail pass :|